include/class.lock.php

```diff
@@ -146,5 +158,22 @@ static function cleanup() {
             'expire__lt' => SqlFunction::NOW()
         ))->delete();
     }
+
+    function validateToken($token=null, $code) {
+        if (!$code)
+            return false;
+
+        $lock = Lock::objects()
+            ->filter(array('code' => $code))
+            ->first();
+
+        if (!$lock)
+            return false;
+
+        if ($token == md5($lock->getId().$lock->getCode().$lock->getExpireTime()))
```

Comment:
Never do this... it spreads hash all over the place (see my other comments). Also I think we should change `signature` to `token`.

```php
if (strcmp($token, $token->getToken()) !== 0)
    return false;

return $lock->delete();

```